extensions/smallrye-health/deployment/src/test/java/io/quarkus/smallrye/health/test/HealthRouteDispatchThreadTest.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.smallrye.health.test;
+
+import java.util.concurrent.Executors;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.smallrye.health.runtime.SmallRyeHealthHandler;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+class HealthRouteDispatchThreadTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(Routes.class, BlockingHandler.class)
+                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
+
+    @Test
+    void testHealthInvokedOnBlockingThread() {
+        RestAssured.given()
+                .when().get("/my-health")
+                .then().statusCode(200);
+
+    }
+
+    @ApplicationScoped
+    static final class Routes {
+
+        public void init(@Observes Router router) {
+            router.get("/my-health").handler(new BlockingHandler());
+            router.get("/my-health").handler(new SmallRyeHealthHandler());
+        }
+    }
+
+    static final class BlockingHandler implements Handler<RoutingContext> {
```

Comment:
Naming: blocking handler is a thing in vertx web. Can you named it OffloadingHandler or ThreadSwitchHandler.